src/components/Approval/Approval.tsx

```diff
@@ -53,11 +67,12 @@ export const Approval = () => {
   } = useWallet()
   const { showErrorToast } = useErrorHandler()
   const { quote, fees } = getValues()
+  const [isExactAllowance] = useWatch({ name: ['isExactAllowance'] })
```

Comment:
Add a `type TS = TradeState<KnownChainIds.EthereumMainnet>`, then we can type `isExactAllowance`:

```suggestion
  const [isExactAllowance] = useWatch({ name: ['isExactAllowance'] }) as [TS['isExactAllowance']]
```

We can then also update line 61 to `} = useFormContext<TS>()`.